fs/mqueue/mq_open.c

```diff
@@ -211,10 +214,15 @@ static int file_mq_vopen(FAR struct file *mq, FAR const char *mq_name,
   /* Make sure that the check for the existence of the message queue
    * and the creation of the message queue are atomic with respect to
    * other processes executing mq_open().  A simple sched_lock() should
-   * be sufficient.
+   * be sufficient for non-SMP case but critical section is needed for
+   * SMP case.
    */
 
+#ifdef CONFIG_SMP
+  flags = enter_critical_section();
+#else
```

Comment:
let' remove shed_lock and always use enter_critical_section?

Reply:
@xiaoxiang781216 

To reduce code maintenance costs, always using enter_critical_setion() would be better. However, using enter_critical_section() for non-SMP cases would increase interrupt latency since it disables interrupts. So sched_lock() would be better for non-SMP cases. What do you think?

Reply:
If the latency is a concern, the better approach is switching to lock by mutex(inode_lock/inode_unlock).

Reply:
BTW, I found that some place under fs/ still use sched_lock/sched_unlock, all should convert to inode_lock/inode_unlock I think.

Reply:
@xiaoxiang781216 

> let' remove shed_lock and always use enter_critical_section?

I will try this one first in this PR.

>BTW, I found that some place under fs/ still use sched_lock/sched_unlock, all should convert to inode_lock/inode_unlock I think.

Then I will study how to remove sched_lock under fs later.